src/client/pkg/tracing/tracing.go

```
@@ -0,0 +1,196 @@
+func InstallJaegerTracerFromEnv() {
+	jaegerOnce.Do(func() {
+		jaegerEndpoint, onUserMachine := os.LookupEnv(jaegerEndpointEnvVar)
+		if !onUserMachine {
+			if host, ok := os.LookupEnv("JAEGER_COLLECTOR_SERVICE_HOST"); ok {
+				port := os.Getenv("JAEGER_COLLECTOR_SERVICE_PORT_JAEGER_COLLECTOR_HTTP")
+				jaegerEndpoint = fmt.Sprintf("%s:%s", host, port)
+			}
+		}
+		if jaegerEndpoint == "" {
+			return // break early -- not using Jaeger
+		}
+
+		// canonicalize jaegerEndpoint as http://<hostport>/api/traces
+		jaegerEndpoint = strings.TrimPrefix(jaegerEndpoint, "http://")
+		jaegerEndpoint = strings.TrimSuffix(jaegerEndpoint, "/api/traces")
+		jaegerEndpoint = fmt.Sprintf("http://%s/api/traces", jaegerEndpoint)
+		cfg := jaegercfg.Configuration{
+			// Configure Jaeger to sample every call, but use the SpanInclusionFunc
+			// addTraceIfTracingEnabled (defined below) to skip sampling every RPC
+			// unless the PACH_ENABLE_TRACING environment variable is set
+			Sampler: &jaegercfg.SamplerConfig{
+				Type:  "const",
+				Param: 1,
+			},
+			Reporter: &jaegercfg.ReporterConfig{
+				LogSpans:            true,
+				BufferFlushInterval: 1 * time.Second,
+				CollectorEndpoint:   jaegerEndpoint,
+			},
+		}
+
+		// configure jaeger logger
+		logger := jaeger.Logger(jaeger.NullLogger)
+		if !onUserMachine {
+			logger = jaeger.StdLogger
+		}
+
+		// Hack: ignore second argument (io.Closer) because the Jaeger
+		// implementation of opentracing.Tracer also implements io.Closer (i.e. the
+		// first and second return values from cfg.New(), here, are two interfaces
+		// that wrap the same underlying type). Instead of storing the second return
+		// value here, just cast the tracer to io.Closer in CloseAndReportTraces()
+		// (below) and call 'Close()' on it there.
+		tracer, _, err := cfg.New(JaegerServiceName, jaegercfg.Logger(logger))
+		if err != nil {
+			panic(fmt.Sprintf("could not install Jaeger tracer: %v", err))
```

I don't really like the idea of panicking if we fail to install Jaeger since it's not a necessary for the core functionality, can we either propagate this error up or just log it in place and have tracing not work if we hit an error here?

---

Done. `IsActive()` just reports if there's a global tracer installed. This now logs an error and then returns before installing a global tracer, so `tracing.IsActive()` returns false and everything downstream works as expected.